omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/scripts/check_token_using_endpoint.lua.erb

```diff
@@ -13,11 +13,12 @@ local res = ngx.location.capture("/organizations/" .. ngx.var.request_org .. "/v
       vars = { upstream = "http://opscode_erchef" }
     })
 
-<% if node['private_chef']['data_collector']['root_url'] && node['private_chef']['data_collector']['token'] -%>
+<% if node['private_chef']['data_collector']['root_url'] && PrivateChef.secrets.exist?('data_collector', 'token') -%>
 if res.status == ngx.HTTP_OK then
   ngx.var.upstream = '<%= URI.parse(node['private_chef']['data_collector']['root_url']).scheme %>://data-collector'
   ngx.req.set_uri('<%= URI.parse(node['private_chef']['data_collector']['root_url']).path %>')
-  ngx.req.set_header("x-data-collector-token", "<%= node['private_chef']['data_collector']['token'] %>")
+  -- Secrets are provided by chef_secrets and will be preloaded into the environment.
+  ngx.req.set_header("x-data-collector-token", os.getenv("DATA_COLLECTOR_TOKEN"))
```

Comment:
I wonder if we can make sure we don't look this up on every request.  It should be cheap, either way though.